database/migrations/20210619211347_dim.sql

```diff
@@ -132,7 +132,8 @@ CREATE TABLE mediafile (
 );
 
 CREATE TABLE users (
-    username TEXT PRIMARY KEY,
+    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
```

Comment:
I wonder if its a good idea to turn all of these changes into proper migrations as running the code behind this PR requires you to delete the DB and start from scratch.

Reply:
I'll take a look at the migration logic tomorrow, but meanwhile:
Adding a file in the migrations folder which transitions the database to the new format while keeping all data should be enough to quietly update all users with older databases?

Reply:
So I took a look at the migrations:
I've written a migration that keeps the user and progress table, but it turns out that the migration in master is incompatible with the migration from the last release.
I believe this is because in commit 2f94d8f9f38fe27c1aba87a9fc0fa5ad844598fa the file [database/migrations/20210619211347_dim.sql](https://github.com/Dusk-Labs/dim/commit/2f94d8f9f38fe27c1aba87a9fc0fa5ad844598fa#diff-2bc028222faea2098fd3a1370a5e96b102408cedf5fac7b5459d87799db5ec45) was updated.

This could be fixed with another migration, but it means that for now, all the migrations are useless.

Reply:
Looks like [database/migrations/20210619211347_dim.sql](https://github.com/Dusk-Labs/dim/commit/91088cbba158ee59cc81b8713fa249e2645abb02#diff-2bc028222faea2098fd3a1370a5e96b102408cedf5fac7b5459d87799db5ec45) was pushed since the last release

Reply:
That migration shouldn't really affect the users much, it was done for performance reasons and could be fixed with a separate PR.

> Adding a file in the migrations folder which transitions the database to the new format while keeping all data should be enough to quietly update all users with older databases?

Yes, I believe this is enough, you can create a new migration with `sqlx-cli`. Since we are changing the primary key of a table, I'm not sure if we can simply drop a column, or add a new column in. So i think we will have to re-create the whole table. I am still unsure if its worth it but it'd be a good idea to get into the habit of thinking about migrations even if dim doesnt have many users yet.

Reply:
I think since the last release was simply a candidate we can get away with breaking backwards compatibility and ship a notice saying that the old database has to be removed. Not ideal but has to be done i guess.

Reply:
I've pushed a commit just now which migrates the `users` and `progress` tables.
It also undoes the two changes in the commits specified above (2f94d8f9f38fe27c1aba87a9fc0fa5ad844598fa and 91088cbba158ee59cc81b8713fa249e2645abb02), and adds them in as a separate migration.
I've tested a migration from the latest release(rc6), and it worked, saving both the users and their progress.

Reply:
Looks good, ended up being simpler than i initially thought.